rng-tools.patch

```diff
@@ -0,0 +1,30 @@
+--- rng-tools.orig	2020-09-20 17:48:03.967820377 +0100
++++ rng-tools	2020-09-20 18:48:01.060123787 +0100
+@@ -6,11 +6,15 @@
+ #
+ ### BEGIN INIT INFO
+ # Provides:		rng-tools
+-# Required-Start:	$remote_fs $syslog
```

Comment:
Also, if rng-tools starts up and doesn't find a syslog, does it completely disable syslog output, or will it output to system once it comes up?

Reply:
pi@box:/opt/iiab/pi-gen $ sudo systemctl status rng-tools
>● rng-tools.service - Start rng-tools
   Loaded: loaded (/etc/systemd/system/rng-tools.service; enabled; vendor preset: enabled)
   Active: active (exited) since Thu 2019-02-14 10:11:59 GMT; 1 years 7 months ago
  Process: 114 ExecStart=/etc/init.d/rng-tools start (code=exited, status=0/SUCCESS)
 Main PID: 114 (code=exited, status=0/SUCCESS)
    Tasks: 4 (limit: 4915)
   CGroup: /system.slice/rng-tools.service
           └─119 /usr/sbin/rngd -r /dev/hwrng
Sep 20 23:20:38 box.lan rngd[119]: stats: FIPS 140-2(2001-10-10) Monobit: 0
Sep 20 23:20:38 box.lan rngd[119]: stats: FIPS 140-2(2001-10-10) Poker: 0
Sep 20 23:20:38 box.lan rngd[119]: stats: FIPS 140-2(2001-10-10) Runs: 0
Sep 20 23:20:38 box.lan rngd[119]: stats: FIPS 140-2(2001-10-10) Long run: 0
Sep 20 23:20:38 box.lan rngd[119]: stats: FIPS 140-2(2001-10-10) Continuous run: 0
Sep 20 23:20:38 box.lan rngd[119]: stats: HRNG source speed: (min=0.000; avg=0.003; max=507704.414)bits/s
Sep 20 23:20:38 box.lan rngd[119]: stats: FIPS tests speed: (min=9.364; avg=12.441; max=20.777)Mibits/s
Sep 20 23:20:38 box.lan rngd[119]: stats: Lowest ready-buffers level: 2
Sep 20 23:20:38 box.lan rngd[119]: stats: Entropy starvations: 0
Sep 20 23:20:38 box.lan rngd[119]: stats: Time spent starving for entropy: (min=0; avg=0.000; max=0)us


pi@box:/opt/iiab/pi-gen $ sudo journalctl -b0 -u rng-tools
>-- Logs begin at Thu 2019-02-14 10:11:59 GMT, end at Sun 2020-09-20 23:26:28 BST. --
Sep 20 20:18:53 box.lan rngd[119]: rngd 2-unofficial-mt.14 starting up...
Sep 20 20:18:53 box.lan rngd[119]: entropy feed to the kernel ready
Sep 20 20:18:53 box.lan rng-tools[114]: Starting Hardware RNG entropy gatherer daemon: rngd.
Sep 20 21:20:38 box.lan rngd[119]: stats: bits received from HRNG source: 80064
Sep 20 21:20:38 box.lan rngd[119]: stats: bits sent to kernel pool: 34336
Sep 20 21:20:38 box.lan rngd[119]: stats: entropy added to kernel pool: 34336
Sep 20 21:20:38 box.lan rngd[119]: stats: FIPS 140-2 successes: 4
Sep 20 21:20:38 box.lan rngd[119]: stats: FIPS 140-2 failures: 0
Sep 20 21:20:38 box.lan rngd[119]: stats: FIPS 140-2(2001-10-10) Monobit: 0
Sep 20 21:20:38 box.lan rngd[119]: stats: FIPS 140-2(2001-10-10) Poker: 0
Sep 20 21:20:38 box.lan rngd[119]: stats: FIPS 140-2(2001-10-10) Runs: 0
Sep 20 21:20:38 box.lan rngd[119]: stats: FIPS 140-2(2001-10-10) Long run: 0
Sep 20 21:20:38 box.lan rngd[119]: stats: FIPS 140-2(2001-10-10) Continuous run: 0
Sep 20 21:20:38 box.lan rngd[119]: stats: HRNG source speed: (min=0.000; avg=0.002; max=506855.217)bits/s
Sep 20 21:20:38 box.lan rngd[119]: stats: FIPS tests speed: (min=9.364; avg=10.888; max=12.450)Mibits/s
Sep 20 21:20:38 box.lan rngd[119]: stats: Lowest ready-buffers level: 2
Sep 20 21:20:38 box.lan rngd[119]: stats: Entropy starvations: 0
Sep 20 21:20:38 box.lan rngd[119]: stats: Time spent starving for entropy: (min=0; avg=0.000; max=0)us
Sep 20 22:20:38 box.lan rngd[119]: stats: bits received from HRNG source: 140064
Sep 20 22:20:38 box.lan rngd[119]: stats: bits sent to kernel pool: 82560
Sep 20 22:20:38 box.lan rngd[119]: stats: entropy added to kernel pool: 82560
Sep 20 22:20:38 box.lan rngd[119]: stats: FIPS 140-2 successes: 7
Sep 20 22:20:38 box.lan rngd[119]: stats: FIPS 140-2 failures: 0
Sep 20 22:20:38 box.lan rngd[119]: stats: FIPS 140-2(2001-10-10) Monobit: 0
Sep 20 22:20:38 box.lan rngd[119]: stats: FIPS 140-2(2001-10-10) Poker: 0

Reply:
> Do other platforms have rng-tools depend on $syslog? E.g. ubuntu

Not sure didn't check.